src/users/users.service.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  BadRequestException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { DeleteResult, Repository } from 'typeorm';
+import { EncryptionService } from '../encryption/encryption.service';
+import { CreateUserDto } from './dto/create-user.dto';
+import { UpdateUserDto } from './dto/update-user.dto';
+import { User } from './entities/user.entity';
+
+@Injectable()
+export class UsersService {
+  constructor(
+    @InjectRepository(User)
+    private readonly userRepository: Repository<User>,
+    private readonly encryptionService: EncryptionService,
+  ) {}
+
+  async create(createUserDto: CreateUserDto) {
+    if (
+      await this.userRepository.findOneBy({ username: createUserDto.username })
```

Comment:
Does it make sense to have the username on a variable to not repeat the call on line 31??